ITTN-066.tex

```diff
@@ -194,6 +194,31 @@ \subsubsection{Discussion}
 As of k8s 1.25, \code{true} is the default value.
 If k8s \code{userns} support reaches ``GA`` in a future release, it is probable that setting \code{hostUsers} to \code{true} will become forbidden.
 
+
+\subsection{Running as Root}\label{sec:nonroot}
+
+Processes of containers SHALL NOT run as root. 
+
+\subsubsection{Discussion}
+
+Exploited applications could allow an attacker to obtain shell access as root. 
+
+\subsection{Disable Service Discovery}\label{sec:discovery}
+
+Pods that do not need to lookup services MUST have this feature disabled. 
+
+\subsubsection{Discussion}
+
+Some pods may not need to lookup or call in-cluster services, this can blocked by not using CodeDNS or not exposing environmental variables. 
+
+\subsection{Read-only Root filesystems}\label{sec:roroot}
+
+Root filesystems of images SHALL be configured as read-only.
```

Comment:
I believe this to require that `readOnlyRootFilesystem` is set to `true` in the `SecurityContext` https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#securitycontext-v1-core of each container within in a pod?

This is yagan as of a few seconds ago:
```bash
 ~ $ k get pods -A -oyaml | grep readOnlyRootFilesystem | sort | uniq -c
    330         readOnlyRootFilesystem: false
     64         readOnlyRootFilesystem: true
```

While I think it is OK to state that it is required policy in this document, I suspect that in practice we would need to use an admission webhook to modify the podSpec to set this.

Reply:
Correct.